pkg/webhook/plugins/nodeaffinitywithcache/node_affinity_with_cache.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2021 The Fluid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeaffinitywithcache
+
+import (
+	"fmt"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/ddc/base"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+/*
+   This plugin is for pods with a dataset.
+   If a pod is labeled for cache, it should require nodes with the mounted dataset on them.
+   else, it should prefer nodes with the mounted dataset on them.
+*/
+
+const NAME = "NodeAffinityWithCache"
```

Comment:
Please change `NAME` to `Name` because we are using  Camel-Case Naming Style. Thanks.

Reply:
fixed, but the other plugins have the same non-camel-case naming style.